File: Algorithm/Python/455. Assign Cookies.py
# -*- coding: utf-8 -*-
# @Author: LC
# @Date:   2016-11-17 23:31:25
# @Last modified by:   WuLC
# @Last Modified time: 2016-11-17 23:31:58
# @Email: liangchaowu5@gmail.com

# sort and two pointers
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        g.sort()
        s.sort()
        pg, ps, count = 0, 0, 0
        while pg < len(g) and ps < len(s):
            if g[pg] <= s[ps]:
                count += 1
                pg += 1
                ps += 1
            else:
                ps += 1
        return count    

File: Algorithm/Java/455. Assign Cookies.java
/*
* @Author: WuLC
* @Date:   2017-06-26 13:35:10
* @Last Modified by:   WuLC
* @Last Modified time: 2017-06-26 13:36:15
* @Email: liangchaowu5@gmail.com
*/

// sort and assign
public class Solution 
{
    public int findContentChildren(int[] g, int[] s) 
    {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0, j = 0, count = 0;
        while (j < s.length && i < g.length)
        {
            if (s[j] >= g[i])
            {
                i++;
                count++;
            }
            j++;
        }
        return count;
    }
}                                                                             